packages/fs.go

```diff
@@ -28,32 +28,47 @@ type PackageFileSystem interface {
 // ExtractedPackageFileSystem provides utils to access files in an extracted package.
 type ExtractedPackageFileSystem struct {
 	path string
+	root fs.FS
 }
 
 func NewExtractedPackageFileSystem(p *Package) (*ExtractedPackageFileSystem, error) {
-	return &ExtractedPackageFileSystem{path: p.BasePath}, nil
+	return &ExtractedPackageFileSystem{path: p.BasePath, root: os.DirFS(p.BasePath)}, nil
 }
 
-func (fs *ExtractedPackageFileSystem) Stat(name string) (os.FileInfo, error) {
-	return os.Stat(filepath.Join(fs.path, name))
+func (fsys *ExtractedPackageFileSystem) Stat(name string) (os.FileInfo, error) {
+	name = strings.TrimPrefix(name, "/")
+	return fs.Stat(fsys.root, name)
 }
 
-func (fs *ExtractedPackageFileSystem) Open(name string) (PackageFile, error) {
-	return os.Open(filepath.Join(fs.path, name))
-}
-
-func (fs *ExtractedPackageFileSystem) Glob(pattern string) (matches []string, err error) {
-	matches, err = filepath.Glob(filepath.Join(fs.path, pattern))
+func (fsys *ExtractedPackageFileSystem) Open(name string) (PackageFile, error) {
+	name = strings.TrimPrefix(name, "/")
+	f, err := fsys.root.Open(name)
 	if err != nil {
-		return
+		return nil, err
 	}
-	for i := range matches {
-		match, err := filepath.Rel(fs.path, matches[i])
+	pf, ok := f.(PackageFile)
```

Comment:
Please add a comment here mentioning that we expect this to be a plain file, that implements this interface. It may look a bit random to convert to this interface otherwise.